### ISSUE TITLE
Getting input from HTTP(s) requests

### ISSUE THREAD

**Issue by ndgnuh** [angusmoore/Matte.jl#43]
This is the kind of use case that:

- You have some kind of worker that send the data from time to time
- Your dashboard listen to the worker, receive those data, process then display it

I can use `update_output` to update the output when the data are received, but how can I receive that kind of input though?

**Comment by angusmoore** [comment]
I haven't tested this particular use case, so there may be some complexities I'm not aware of. But off the top of my head, this use case is no different whether you're using Matte or not (other than the `update_output` of course). All you'll need to do is create a function in the server module to query the data source repeatedly (e.g. via HTTPs request), with some `sleep` in between each request so you aren't flooding the server, in an infinite loop (or a loop with some controllable shutdown condition if you like). Then just start that function as an async task (i.e. using julia `@async` macro) so that it is non-blocking for anything else the server needs to do.

I haven't tested this particular use case, but I think something along those lines should work. Hope that helps, let me know how you go.

**Comment by ndgnuh** [comment]
Thanks, I'll try that out to see if that works. Polling sounds a bit unappealing, but that's just my feeling though. Maybe I'll open another server to listen to incoming data and find a way to make the two servers communicate.

**Comment by angusmoore** [comment]
Yes, fair. I think you could equally set up a lightweight server inside the Matte server module (e.g. using `HTTP.jl` or stdlib `Sockets`) and have that listener handle incoming data in the same way. Provided that listener is running in `@async`, I _think_ it should be ok. But it's not something I've tested, so there may be some complication I'm missing.
